addons/WritingCalculations/src/presenter.js

```diff
@@ -1080,22 +1080,18 @@ function AddonWritingCalculations_create() {
     };
 
     presenter.reset = function() {
-        this.clean(true, true);
-        const $emptyBoxesInputs = presenter.getEmptyBoxesInputs();
-        if (typeof(presenter.userAnswers) !== "undefined") {
-            $.each($emptyBoxesInputs, function(index){
-                presenter.userAnswers[index] = '';
```

Comment:
To była przyczyna błędu. Powodowała że w presenter.userAnswers były puste wartości a cała lista powinna być pusta. SA i GSA operuje na mechanizmie push aby zapisać odpowiedzi użytkownika, więc jak tutaj było coś dodawane to brało niepoprawne dane.